simulator/mips/mips_instr.h

```diff
@@ -89,39 +121,59 @@ class BaseMIPSInstr
         Trap trap = Trap::NO_TRAP;
 
         // Endian specific
-        const union _instr
+        const struct _instr
         {
-            const struct AsR
+            struct AsR
             {
-                uint32 funct  :6;
-                uint32 shamt  :5;
-                uint32 rd     :5;
-                uint32 rt     :5;
-                uint32 rs     :5;
-                uint32 opcode :6;
-            } asR;
-            const struct AsI
+                int funct;
+                int shamt;
```

Comment:
You have to use `uint32` here, as it matches semantics and has a well defined sizeof